web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.tsx

```diff
@@ -72,60 +130,171 @@ export function TestConnection(props: AgentStepProps) {
+              <StepSkeletonPickUser>
+                <Box width="320px">
+                  <LabelInput>Select Login</LabelInput>
+                  <Select
+                    value={selectedLoginOpt}
+                    options={sortNodeLogins(fetchLoginsAttempt.data).map(
+                      mapLoginToSelectOption
+                    )}
+                    onChange={(o: Option) => setSelectedLoginOpt(o)}
+                    isDisabled={connectionDiagAttempt.status === 'processing'}
+                  />
+                </Box>
+              </StepSkeletonPickUser>
+            )}
+            <ConnectionDiagnosticResult
+              attempt={connectionDiagAttempt}
+              diagnosis={diagnosis}
+              canTestConnection={canTestConnection}
+              testConnection={() => testConnection(selectedLoginOpt.value)}
+              stepNumber={1 + stepOffset}
+              stepDescription="Verify that your computer is accessible"
+              numberAndDescriptionOnSameLine
+            />
+            <StyledBox>
+              <Text bold mb={3}>
+                Step {2 + stepOffset}: Connect to Your Computer
+              </Text>
+              <ButtonSecondary
+                width="200px"
+                onClick={() => startSshSession(selectedLoginOpt.value)}
+              >
+                Start Session
+              </ButtonSecondary>
+            </StyledBox>
+          </>
+        ))}
+      <ActionButtons
+        disableProceed={
+          fetchLoginsAttempt.status !== 'success' ||
+          fetchLoginsAttempt.data.length === 0
+        }
+        onProceed={nextStep}
+        lastStep={true}
+        // onPrev is not passed on purpose to disable the back button. The flow would go back to
+        // polling which wouldn't make sense as the user has already connected their computer so the
+        // step would poll forever, unless the user removed the agent and configured it again.
       />
-      <StyledBox>
-        <Text bold>Step 3</Text>
-        <Text typography="subtitle1" mb={3}>
-          Connect to the server
-        </Text>
-        <ButtonSecondary
-          width="200px"
-          onClick={() => startSshSession(selectedOpt.value)}
-        >
-          Start Session
-        </ButtonSecondary>
-      </StyledBox>
-      <ActionButtons onProceed={nextStep} lastStep={true} onPrev={prevStep} />
     </Box>
   );
 }
+
+const StepSkeletonPickUser = (props: { children: React.ReactNode }) => (
+  <StyledBox mb={5}>
+    <Text bold mb={3}>
+      Step 1: Pick the OS user to test
+    </Text>
+    {props.children}
+  </StyledBox>
+);
+
+/**
+ * To display the first step, fetchLoginsAttempt must be successful, hence why we show errors from
+ * fetchLoginsAttempt within the first step.
+ */
+const FetchLoginsAttemptError = (props: {
+  children: React.ReactNode;
+  buttonText: string;
+  buttonOnClick: () => void;
+}) => (
+  <StepSkeletonPickUser>
+    <>
```

Comment:
Unnecessary `<>`.

Reply:
There was a few more unnecessary fragments in this file. react/jsx-no-useless-fragment doesn't catch that, even with `allowExpressions` disabled.